features/vaadin-dashlets/dashlet-ksc/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/KscDashlet.java

```diff
@@ -430,6 +423,13 @@ public Component getComponent() {
         });
     }
 
+    String determineNodeIdByResourceId(ResourceId resourceId){
+        OnmsResource parentResource = m_resourceDao.getResourceById(resourceId).getParent();
```

Comment:
The parent can be null, in which case the we should attempt to retrieve the node directly from the given resource.